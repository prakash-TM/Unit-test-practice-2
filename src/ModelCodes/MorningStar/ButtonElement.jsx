import React from 'react';

const ButtonElement = (props)=>{

    return (
        <>
        <p title="Para tab">My name is {props.label}</p>
        <button title="This is a new button" onClick={()=>{console.log('message')}}>{props.label}</button>
        </>
    )
}

export default ButtonElement