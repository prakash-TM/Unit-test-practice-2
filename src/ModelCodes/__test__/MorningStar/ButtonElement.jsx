import React from 'react';
import ButtonElement from "../../MorningStar/ButtonElement";
import { create } from 'react-test-renderer';

describe('Button loading correctly', () => {
  test('Button loading correctly', () => {
    const button = create(<ButtonElement />);
    //console.log(button);

    expect(button.toJSON()).toBeTruthy();
  });

  test('Button should be having label prop', ()=>{
    const button = create(<ButtonElement name="Ashwary" label="Click me"/>);

    const {root } = button;
    const {props} = root

    expect(props).not.toEqual({});
    expect(props.label).toBeTruthy();
    expect(button.toJSON()).toMatchSnapshot();
  })
});