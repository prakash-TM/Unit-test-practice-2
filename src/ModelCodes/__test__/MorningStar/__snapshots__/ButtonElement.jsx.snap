// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button loading correctly Button should be having label prop 1`] = `
Array [
  <p
    title="Para tab"
  >
    My name is 
    Click me
  </p>,
  <button
    onClick={[Function]}
    title="This is a new button"
  >
    Click me
  </button>,
]
`;
